[Caitlyn Tang] iP

caitlyntang opened the pull request:
## Anxi the Chatbox 

>"Believe you can and you're halfway there." - Theodore Roosevelt ([source](https://www.usatoday.com/story/life/2023/11/30/positive-quotes-to-inspire/11359498002/))

Anxi frees your mind of having to remember things you need to do. It's

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :star_struck: 

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

caitlyntang committed:
complete renaming and adding skeleton version of chatbot

caitlyntang committed:
add basic user input functionality

caitlyntang committed:
add item list functionality

caitlyntang committed:
add ability to mark tasks as done, add new task class

caitlyntang committed:
change implementation for reading in user inputs

caitlyntang committed:
added categories and new classes for different tasks, update how task is printed in main

caitlyntang committed:
add test cases, fix typo

caitlyntang committed:
add error handling

caitlyntang committed:
add test cases, fix error handling bug

caitlyntang committed:
add delete functionality with error handling, add error handling for mark and unmark

caitlyntang committed:
Revert Duke logo

caitlyntang committed:
Add save class

caitlyntang committed:
Move file access error handling to seperate function

caitlyntang committed:
Implement loading tasks from storage

caitlyntang committed:
Save new entries to file

caitlyntang committed:
Add delete function for saving to hard disk

caitlyntang committed:
Add function to update task completion in hard disk

caitlyntang committed:
Fix minor bug writing to file

caitlyntang committed:
Merge branch 'branch-Level-7'

caitlyntang committed:
Store dates using java.time

caitlyntang committed:
Add some compatability with different date formats

caitlyntang committed:
Merge branch 'branch-Level-8'

dayn-1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+
+        String action = br.readLine();
+        StringTokenizer st = new StringTokenizer(action, " ");
+        String temp = st.nextToken();
```
Maybe variable ```temp``` can be given a more specific name

dayn-1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+                            throw new DukeException();
+                        }
+
+                        int u = Integer.parseInt(st.nextToken());
```
Maybe ```u``` can be ```updateIndex``` or equivalent instead.

dayn-1 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.List;
+
+public class Event extends Task {
+    protected LocalDateTime from;
```
Perhaps ```startDate``` or something like that will be more clear

dayn-1 commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,53 @@
+        this.to = LocalTime.parse(to, DateTimeFormatter.ofPattern("HHmm"));
+    }
+
+    public Event (String description, boolean done, String from, String to) {
```
Maybe ```isDone``` instead of ```done```

dayn-1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,193 @@
+                            throw new DukeException();
+                        }
+
+                        int del = Integer.parseInt(st.nextToken());
```
Maybe ```deleteIndex``` may work better?

dayn-1 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+        this.isDone = isDone;
+    }
+
+    public String getSaveTask() {
```
Maybe this can be renamed? Not sure what it means

dayn-1 reviewed (commented):
Overall good work! Just a suggestion to use more variables or variables with more specific names to make the code more readable.

caitlyntang committed:
Consolidate parseDate function into Task class

caitlyntang committed:
Update command processing to take in more date time formats

caitlyntang committed:
Resolve save to file bug, change date time parsing method

caitlyntang committed:
Merge branch 'branch-Level-8'

caitlyntang committed:
Initial implementation refactoring code to OOP

caitlyntang committed:
Fix save delete to file bug

caitlyntang committed:
Delete duplicate save class file

caitlyntang committed:
Diversify accepted time formats in event class

caitlyntang committed:
Change naming for parseDate, remove extra whitespace

caitlyntang committed:
Merge branch 'branch-OOP'

caitlyntang committed:
Seperate classes into packages

caitlyntang committed:
Add error message for missing event details

caitlyntang committed:
Merge branch 'add-gradle-support'

caitlyntang committed:
Add JUnit tests for deadline, event and todo tasks

caitlyntang committed:
Merge branch 'branch-JUnit'

caitlyntang committed:
Add archiveFileName to build.gradle

caitlyntang committed:
Add find functionaliity

caitlyntang committed:
Merge branch 'branch-Level-9'

caitlyntang committed:
Merge branch 'master' into branch-A-CodingStandard

caitlyntang committed:
Update naming convention in task, event, deadline and todo

caitlyntang committed:
Update naming conventions, remove unneeded imports

caitlyntang committed:
Change switch statement format

caitlyntang committed:
Update naming, remove unused function

caitlyntang committed:
Merge branch 'master' into branch-A-JavaDoc

caitlyntang committed:
Add java docs for deadline, event, todo and task

caitlyntang committed:
Add java docs for remaining files

caitlyntang committed:
Merge branch 'branch-A-CodingStandard'

caitlyntang committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/command/Parser.java
#	src/main/java/duke/command/Storage.java
#	src/main/java/duke/command/Ui.java
#	src/main/java/duke/tasks/Event.java
#	src/main/java/duke/tasks/Task.java

caitlyntang committed:
Remove debugging print statement

caitlyntang committed:
Add checkstyle files

caitlyntang committed:
Merge branch 'branch-A-Checkstyle'

caitlyntang committed:
Fix checkstyle errors

caitlyntang committed:
Set up javaFx

caitlyntang committed:
Add basic GUI for chatbox

caitlyntang committed:
Integrate GUI into Duke

caitlyntang committed:
Resolve checkstyle errors

caitlyntang committed:
Update javadocs

caitlyntang committed:
Merge branch 'branch-Level-10'

caitlyntang committed:
Add assertions for file creation, write and delete

caitlyntang committed:
Split parseInput into classes for event handling

caitlyntang committed:
Move files related to GUI to display package

caitlyntang committed:
Refactor function start in Duke

The start function is more than 30 lines of code. Which decreases the
readability of the function.

Abstract start function into 3 seperate functions, 1 for each stage of
initialization.

This splits the steps of initiailzing the GUI into 3 distinct stages for
easier changes in future.

caitlyntang committed:
Update javadocs, remove redundant method

caitlyntang committed:
Bug fix write to file during delete
Update gradle path

caitlyntang committed:
Update javadocs to meet coding standard

caitlyntang committed:
Merge pull request #2 from caitlyntang/branch-A-Assertions

Add assertions

caitlyntang committed:
Merge branch 'master' into branch-A-CodeQuality

caitlyntang committed:
Merge pull request #3 from caitlyntang/branch-A-CodeQuality

Improve code quality

caitlyntang committed:
Seperate handllers into different package

caitlyntang committed:
Add new TimeHandler class

Time handling functions are spread across multiple classes. Some
classes share the time handling functions.

This makes it hard to debug as it is hard to keep track of the multiple
time handling functions across the different files.

Extract the different time functions into a dedicated class.

Consolidating the location of time related functions makes it easy to
scale and debubg if there are any errors related to these functions.

caitlyntang committed:
Add view tasks on specific date

caitlyntang committed:
Merge branch 'branch-B-ViewSchedules'

caitlyntang committed:
Print welcome message, change taskbot images

caitlyntang committed:
Update output string formats

caitlyntang committed:
Remove dead code

caitlyntang committed:
Make layout resizable

caitlyntang committed:
Add program exit handling

caitlyntang committed:
Merge branch 'branch-A-BetterGui'

caitlyntang committed:
Add error handling for parsing date and time

caitlyntang committed:
Add additional message for find command no matching task

caitlyntang committed:
Bug fix storage

caitlyntang committed:
Merge branch 'A-MoreErrorHandling'

caitlyntang committed:
Rename duke package to anxi

caitlyntang committed:
Change filepath, update responses

caitlyntang committed:
Merge branch 'branch-A-Personality'

caitlyntang committed:
Update existing JUnit tests

caitlyntang committed:
Merge branch 'branch-A-MoreTesting'

caitlyntang committed:
Add text wrapping for input commands, change unknown command responce

caitlyntang committed:
Add Ui.png

caitlyntang committed:
Update README.md

caitlyntang committed:
Add sample data file

caitlyntang committed:
Fix ui not showing in README

caitlyntang committed:
Update gradle configurations

caitlyntang committed:
Improve error handling

caitlyntang committed:
Merge master into branch

caitlyntang committed:
Merge branch 'master' into branch-A-MoreErrorHandling

caitlyntang committed:
Replace TimeHandler with a more generic Handler class
Make necessary changes to the child classes of Handler

caitlyntang committed:
Improve error logging for exceptions, delete dead functions

caitlyntang committed:
Merge branch 'branch-A-MoreErrorHandling'

caitlyntang committed:
Improve output comments, fix typo

caitlyntang committed:
Standardise accepted date formats, add more robust error messages

caitlyntang committed:
Change accepted date formats

caitlyntang committed:
Fix date parser error

caitlyntang committed:
Fix error comment for parseDate
